miio/gateway.py

```diff
@@ -207,6 +224,47 @@ def discover_devices(self):
 
         return self._devices
 
+    def x_del(self, script_id):
+        """Delete script by id."""
+        return self.send("miIO.xdel", [script_id])
+
+    @command(
+        click.argument("sid"),
+        click.argument("command"),
+        click.argument("encoded_token"),
+    )
+    def subdevice_command(self, sid, command, encoded_token):
+        """Send command to subdevice."""
+        self.discover_devices()
+        target = list(filter(lambda subdevice: subdevice.sid == sid, self.devices))
+        if len(target) < 1:
+            return f"Device with sid={sid} not found"
+        elif not hasattr(target[0], command):
+            return f"Device with sid={sid} has no method {command}"
+        else:
+            return getattr(target[0], command)(encoded_token)
+
+    def install_script(self, sid, builder, encoded_token, ip=None):
+        """Install script for by building script source and sending it with miio method. You need to run fake or real device to capture script execution results."""
+        if ip is None:
```

Comment:
I'd like to see this being much simpler, in my opinion the API should be more something like:
```
gw = Gateway()
cube = gw.get_device_by_id("cube id")
some_other_device = gw.get_device_by_id("my light")
cube.install_event_handler(cube, 'shake', some_other_device, 'some action')
```

Likewise, if I have a cube instance, I would like to query the list of available events and already installed scripts:
```
print("available events: %s" % cube.events)  # I mention this way of exposing the available events in some other comment

# if there is only one script possible, otherwise we need a way to get a list of already installed scripts to make it possible for the user to decide which one to remove
cube.remove_script("shake")  

# the other case
handlers = cube.installed_event_handlers
for handler in handers:
    print("handler %s")
```